gemini/sample-apps/fashion-trends-prediction/app/data_prep/dataProcessing.py

```diff
@@ -0,0 +1,378 @@
+    ).text
+
+    try:
+        response = json.loads(response)
+    except Exception as e:
+        print(e)
+        return {}
+
+    for json_ob in response:
+        for category in json_ob.keys():
+            if category in answer:
+                answer[category].append(json_ob[category])
+            else:
+                answer[category] = [json_ob[category]]
+
+    return answer
+
+
+def generateCaptionImageQnA(image_path):
+    """Generates a caption for an image using the ImageQnA model.
+
+    Args:
+                    image_path (str): The path to the image file.
+
+    Returns:
+                    dict: A dictionary containing the generated caption.
+    """
+
+    answer = {}
+
+    # Load the downloaded image
+    user_image = Image.load_from_file(image_path)
+
+    if (
+        str(
+            image_qna_model.ask_question(
+                image=user_image,
+                question=qList[0],
+                number_of_results=1,
+            )[0]
+        )
+        == "no"
+    ):
+        return answer
+
+    items = str(
+        image_qna_model.ask_question(
+            image=user_image,
+            question=qList[1],
+            number_of_results=1,
+        )[0]
+    ).split(",")
+
+    for item in items:
+        res = set()
+        for q in qList2:
+            res.add(
+                str(
+                    image_qna_model.ask_question(
+                        image=user_image,
+                        question=q.replace("x", item),
+                        number_of_results=1,
+                    )[0]
+                )
+            )
+
+        result = " ".join(res)
+
+        answer[item] = [result]
+
+    answer["Sleeves"] = [
+        str(
+            image_qna_model.ask_question(
+                image=user_image,
+                question=qList[2],
+                number_of_results=1,
+            )[0]
+        )
+    ]
+
+    answer["Neckline"] = [
+        str(
+            image_qna_model.ask_question(
+                image=user_image,
+                question=qList[3],
+                number_of_results=1,
+            )[0]
+        )
+    ]
+
+    answer["Jewellery"] = [
+        str(
+            image_qna_model.ask_question(
+                image=user_image,
+                question=qList[4],
+                number_of_results=1,
+            )[0]
+        )
+    ]
+
+    return answer
+
+
+def generateCaption(image_path, model):
+    """Generates a caption for an image using the specified model.
+
+    Args:
+                    image_path (str): The path to the image file.
+                    model (str): The model to use for generating the caption.
+
+    Returns:
+                    dict: A dictionary containing the generated caption.
+    """
+
+    if model == "Gemini":
+        return generateCaptionGemini(image_path)
+
+    elif model == "ImageQnA":
+        return generateCaptionImageQnA(image_path)
+
+
+def getPosts(user, previous, cnt=10, cookies={}, model="Gemini"):
```

Comment:
Don't pass an empty dict as a default argument. Can lead to side effects. Set default as `None`